### PR TITLE
Make sure antialiasing is set right during context creation

### PR DESCRIPTION
**Pull request by sankhesh**
Fixes issue #1400

### CODE REVIEW

**Comment by sankhesh** [comment]
@dlonie Please review

File: Packages/vcs/Lib/VTKPlots.py [hunk 1]
```diff
@@ -293,7 +295,7 @@ def createRenWin(self,*args,**kargs):
       self.renWin.SetStencilCapable(1)
       ## turning off antialiasing by default
       ## mostly so that pngs are same accross platforms
```
Comment (dlonie) [inline Packages/vcs/Lib/VTKPlots.py]: Let's remove this comment, since it's no longer true after this patch.

File: Packages/vcs/Lib/VTKPlots.py [hunk 2]
```diff
@@ -52,6 +52,8 @@ def __init__(self,canvas,renWin=None, debug=False,bg=None):
     self.numberOfPlotCalls = 0
     self.renderWindowSize=None
     self.clickRenderer = None
+    # Turn off anti-aliasing by default
+    self.antiAliasing = 0
```
Comment (dlonie) [inline Packages/vcs/Lib/VTKPlots.py]: Let's just call it `self.antialiasing`, all lowercase (to match the set/get calls)

**Comment by dlonie** [comment]
A couple minor issues, but looks good to me otherwise :+1:

**Comment by sankhesh** [comment]
Implemented @dlonie's suggestions in e1b093d

**Comment by aashish24** [comment]
Nice!! this will make things look much better. We should send an email once this branch is merged with before and after (pick a crappy looking plot from before)

**Comment by dlonie** [comment]
Works here! @doutriaux1 Want to test/merge as well? @sankhesh and I are both at Kitware.

**Comment by dlonie** [comment]
@aashish24 Here are the images I sent you and Brian while investigating this. At full resolultion it's quite an improvement.

**No multisampling (current master)**
orig

**16x MSAA (possible after this patch)**
with_msaa

**16x MSAA + `gm.linewidth = 2` (best match to Brian's sample plots)**
with_msaa_linewidth2

**Comment by aashish24** [comment]
Nice! is 16x MSAA is set by you on your driver or someplace else?

**Comment by dlonie** [comment]
The argument to `Canvas.setantialiasing` (and eventually `vtkRenderWindow::SetMultiSamples`) is the number of samples to take.

So after this patch, users can do:

```
x = vcs.init()
x.setantialiasing(16) # or 8, or 4, or 2, or 0...
x.plot(...)
```

and the result will be antialiased. If the graphics hardware/driver doesn't support the number of requested samples, VTK will set it as high as possible for the hardware/driver.

**Comment by aashish24** [comment]
Okay, sounds good to me. Thanks. @doutriaux1, some folks want this today as they producing 1D plots. Would it be possible for you to give it a thumbs up?

**Comment by jypeter** [comment]
@aashish24, can you try running the script in #1079 and see if you get nice(r) dots and markers, when using the smoothing?

**Comment by doutriaux1** [comment]
great! building now and will merge.

**Comment by aashish24** [comment]
@jypeter I will request @sankhesh to do it. @sankhesh can you post your results here?

**Comment by doutriaux1** [comment]
@sankhesh k it works and it's great. Unfortunately for you it's way too great to be left to the user. So can I please ask you to tweak this so that antialiasing is on by default. Of course that means turning it off for all vcs tests... sorry.

**Comment by aashish24** [comment]
@doutriaux1 we cannot turn it ON for testing since it will make resulting image vary depending on the user's setting. I would suggest we merge this one in and have @sankhesh push another branch where we turn it OFF for testing and having it ON by default.

**Comment by aashish24** [comment]
Ooops..sorry I just read your email wrong. I totally agree with you and that's what we were planning to do it (as we do the same for VTK).

**Comment by doutriaux1** [comment]
@aashish24 should we open another issue for this then? I was hoping @sankhesh would stick it in this branch.

**Comment by doutriaux1** [comment]
done, see: https://github.com/UV-CDAT/uvcdat/issues/1433

**Comment by sankhesh** [comment]
@jypeter Here is a montage of screenshots generated using the script in #1079 using different antialiasing modes:
screenshot from 2015-06-29 11 41 58

No antialiasing:
test_simple_marker_no_msaa

8x MSAA:
test_simple_marker_8x_msaa

16x MSAA:
test_simple_marker

**Comment by jypeter** [comment]
Thanks @sankhesh , it does look nicer with 8x and 16x (8x may be enough)!

I don't remember if I have already asked this, but is it possible to generate nicer circles and even triangles from the start (before the smoothing stage)? When you look at the triangle in the "no MSAA" picture above, it looks pixelized, but with much bigger pixels than the pixels in the "trian" string

**Comment by doutriaux1** [comment]
@sankhesh it's probably just a matter of tweaking the vtksource object we're using.

**Comment by aashish24** [comment]
@doutriaux1 yes, that should be possible. I will create a new issue for it.

**Pull request merged**